apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -45,9 +44,10 @@ internal ClientPool(ISpannerClientFactory clientFactory = null)
 
         public async Task<SpannerClient> AcquireClientAsync(
             ITokenAccess credentials = null,
-            ServiceEndpoint endpoint = null)
+            ServiceEndpoint endpoint = null,
+            IDictionary additionalOptions = null)
```

Comment:
in fact, I'm more likely to eliminate it all and just use the connectionstringbuilder itself as the key.  then, it can override equals as the full formed normalized connectionstring (plus the credentials I guess).